.github/workflows/deno.yml

```
@@ -1,44 +1,59 @@
-name: Node.js
+name: Deno
 
 on:
   push:
   pull_request:
   schedule:
-    # Check if it works with current dependencies
+    # Check if it works with current versions
     - cron: '42 2 * * 6' # weekly on Saturday 2:42 UTC
 
 jobs:
+  denofmt-and-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: denoland/setup-deno@v1
+        with:
+          deno-version: v1.x
+      - uses: actions/checkout@v4
+
+      - run: deno lint
+      - run: deno fmt --check
+
   test:
-    name: Node.js
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/setup-node@v4
+      - uses: denoland/setup-deno@v1
         with:
-          node-version: 'lts/*'
+          deno-version: v1.x
       - uses: actions/checkout@v4
-      - run: npm install
-      - run: npm test
-      - run: npm pack
 
-  publish:
-    name: Publish on NPM
-    if: startsWith(github.ref, 'refs/tags/v')
+      - run: deno cache source/*.ts
+      - run: deno check source/*.ts
+      - run: deno test
+
+  npm:
+    name: Node.js Package
     runs-on: ubuntu-latest
-    needs: test
     permissions:
       contents: write
       id-token: write
     steps:
       - uses: actions/checkout@v4
+      - uses: denoland/setup-deno@v1
+        with:
+          deno-version: v1.x
       - uses: actions/setup-node@v4
         with:
           node-version: 'lts/*'
           registry-url: 'https://registry.npmjs.org'
       - run: npm install
+      - run: npm pack
       - run: npm publish
```

This runs `"prepare"` 3 times.
```suggestion
      - run: npm install --ignore-scripts
      - run: npm run prepare
      - run: npm publish --ignore-scripts
```

---

`npm publish` already lists packed files.

---

`npm publish` is only executed on tags. `pack` always tests that the packaging works and displays what it contains.

---

On the normal run `prepare` is run 2 times but I think its cleaner to run `npm install` and `npm pack` without arguments just like users would. deno2node is fast so its not much of a downside.